src/compiler/emitter.ts

```diff
@@ -300,14 +300,29 @@ namespace ts {
                             continue;
                         }
 
-                        writeLines(helper.text);
+                        writeHelper(writeLines, helper.text, makeUniqueName);
                         helpersEmitted = true;
                     }
                 }
             }
 
             return helpersEmitted;
         }
+
+        function writeHelper(writer: (text: string) => void, helper: EmitHelper["text"], makeUniqueName: UniqueNameHandler) {
+            if (typeof helper === "string") {
+                writer(helper);
+            }
+            else {
+                writer(helper({
+                    makeFileLevelUniqueName: name => makeUniqueName(name, isFileLevelUniqueName, /*optimistic*/ true),

```

Comment:
This could be pulled out into a function to avoid allocating a new function object every time this is called.